src/CommandLineParser.ts

```diff
@@ -0,0 +1,185 @@
+import { CLI_ERROR } from './Errors';
+
+import yargs from 'yargs';
+import colors from 'colors';
+import fs from 'fs';
+
+export const argv = yargs.options({
+    videoUrls: {
```

Comment:
Just a heads up that i'm going to --kebab-case the bjesus out of these args, but right before we officially tag it v2.0.